server-base/src/main/java/org/apache/kylin/rest/service/JobService.java

```diff
@@ -435,6 +438,26 @@ private void checkAllowBuilding(CubeInstance cube) {
         }
     }
 
+    private void checkStreamLambdaBuildingSegment(CubeInstance cube, TSRange tsRange) {
+        Segments<CubeSegment> readycubeSegments = cube.getSegments(SegmentStatusEnum.READY);
+        if (readycubeSegments.size() <= 0) {
```

Comment:
I suggest you to remove the branch: `readycubeSegments.size() <= 0` .

Reply:
Done, remove this branch.